src/main/java/de/wuespace/telestion/extension/mongodb/DataService.java

```diff
@@ -72,106 +48,55 @@ public Configuration getConfig() {
 	private final String dbSave = Address.incoming(MongoDatabaseService.class, "save");
 	private final String dbFind = Address.incoming(MongoDatabaseService.class, "find");
 
-	/**
-	 * Method to register consumers to the eventbus.
-	 */
-	private void registerConsumers() {
-		vertx.eventBus().consumer(inFind, request -> {
-			JsonMessage.on(DataRequest.class, request, req -> {
-				this.dataRequestDispatcher(req, res -> {
-					if (res.failed()) {
-						logger.error(res.cause().getMessage());
-						request.fail(-1, res.cause().getMessage());
-						return;
-					}
-					request.reply(res.result());
-				});
-			});
-		});
-		vertx.eventBus().consumer(inSave, document -> {
-			JsonMessage.on(JsonMessage.class, document, doc -> {
-				vertx.eventBus().request(dbSave, doc.json(), res -> {
-					if (res.failed()) {
-						logger.error(res.cause().getMessage());
-						document.fail(-1, res.cause().getMessage());
-						return;
-					}
-					document.reply(res.result().body());
-				});
-			});
-		});
-	}
-
 	/**
 	 * Parse and dispatch incoming DataRequests.
-	 *
-	 * @param request			Determines which dataType should be retrieved and if an Operation should be executed.
-	 * @param resultHandler		Handles the request to the underlying database. Can be failed or succeeded.
 	 */
-	private void dataRequestDispatcher(DataRequest request, Handler<AsyncResult<JsonObject>> resultHandler) {
-		if (request.operation().isEmpty()) {
-			this.fetchLatestData(request.collection(), request.query(), res -> {
-				if (res.failed()) {
-					resultHandler.handle(Future.failedFuture(res.cause().getMessage()));
-					return;
-				}
-				resultHandler.handle(Future.succeededFuture(res.result()));
-			});
-		} else {
-			var dataOperation = new DataOperation(new JsonObject(), request.operationParams());
-			this.fetchLatestData(request.collection(), request.query(), res -> {
-				if (res.failed()) {
-					resultHandler.handle(Future.failedFuture(res.cause().getMessage()));
-					return;
-				}
-				dataOperation.data().put("data", res.result());
+	private void dataRequestDispatcher(DataRequest request, Message<JsonObject> message) {
+		var latestData = fetchLatestData(request.collection(), request.query());
+
+		// manipulate latest data if request is more specific
+		if (!request.collection().isEmpty()) {
+			latestData = latestData.compose(result -> {
```

Comment:
Here, the returned future from `compose` is from interest, not the actual latest data.